src/value.rs

```diff
@@ -241,6 +319,172 @@ impl Value {
         }
     }
 
+    /// Returns `self` into an i128, if possible.
+    pub fn into_int128(self) -> Result<i128> {
+        match self.kind {
+            ValueKind::I64(value) => Ok(value as i128),
+            ValueKind::I128(value) => Ok(value),
+            ValueKind::U64(value) => Ok(value as i128),
+            ValueKind::U128(value) => Err(ConfigError::invalid_type(
+                self.origin,
+                Unexpected::U128(value),
+                "an signed 128 bit integer",
+            )),
+
+            ValueKind::String(ref s) => {
+                match s.to_lowercase().as_ref() {
```

Comment:
Why do we do this, but don't transparently allow, say, `u128` values that are in range of `i128` to convert to `i128`?

Reply:
I guess you wanted another line to be commented here? I will investigate where we can do this, yes!

Reply:
No, I meant to ask why we transparently convert strings like `yes` or `no` to integers, but not integers to other integers. Seemed somewhat counterintuitive.

Reply:
Ah, now I see what you mean.

This patch tries to mimic the behavior of the non-int128 code... I agree that this might be not what we want for the next release, but I would introduce a change to that behavior in another PR, not in this one.